AFQ/data.py

```diff
@@ -288,6 +295,1039 @@ def read_templates(resample_to=False):
+        non_sub_s3_keys = {}
+
+        s3_prefix = '/'.join([self.bucket, self.s3_prefix])
+
+        nonsub_keys = _ls_s3fs(s3_prefix=s3_prefix,
+                               anon=self.anon)['other']
+        nonsub_keys = [k for k in nonsub_keys
+                       if not k.endswith('derivatives')]
+
+        nonsub_deriv_keys = []
+        for dt in self.derivative_types:
+            s3_prefix = '/'.join([
+                self.bucket,
+                self.s3_prefix,
+                'derivatives',
+                dt
+            ])
+
+            nonsub_deriv_keys.append(_ls_s3fs(
+                s3_prefix=s3_prefix,
+                anon=self.anon
+            )['other'])
+
+        non_sub_s3_keys = {
+            'raw': nonsub_keys,
+            'derivatives': nonsub_deriv_keys,
+        }
+
+        return non_sub_s3_keys
+
+    def _list_all_subjects(self):
+        """Return list of subjects
+
+        Returns
+        -------
+        list
+            list of participant_ids
+        """
+        if self._use_participants_tsv:
+            tsv_key = '/'.join([self.s3_prefix, 'participants.tsv'])
+            s3 = get_s3_client(anon=self.anon)
+
+            def get_subs_from_tsv_key(s3_key):
+                response = s3.get_object(
+                    Bucket=self.bucket,
+                    Key=s3_key
+                )
+
+                return set(pd.read_csv(
+                    response.get('Body')
+                ).participant_id.values)
+
+            subject_set = get_subs_from_tsv_key(s3_key)
+            subjects = list(subject_set)
+        else:
+            s3_prefix = '/'.join([self.bucket, self.s3_prefix])
+            sub_keys = _ls_s3fs(s3_prefix=s3_prefix,
+                                anon=self.anon)['subjects']
+
+            # Just need BIDSLayout for the `parse_file_entities`
+            # method so we can pass dev/null as the argument
+            layout = BIDSLayout(os.devnull, validate=False)
+            subjects = []
+            for key in sub_keys:
+                entities = layout.parse_file_entities(key)
+                subjects.append('sub-' + entities.get('subject'))
+
+        return subjects
+
+    def _download_non_sub_keys(self, directory, select="all"):
+        fs = s3fs.S3FileSystem(anon=self.anon)
+        for fn in self.non_sub_s3_keys['raw']:
+            if select == "all" or any([s in fn for s in select]):
+                fs.get(fn, op.join(directory, op.basename(fn)))
+
+    def download(self, directory, include_modality_agnostic=False,
+                 include_dataset_description=True,
+                 include_derivs=False, overwrite=False, pbar=True):
+        """Download files for each subject in the study
+
+        Parameters
+        ----------
+        directory : str
+            Directory to which to download subject files
+
+        include_modality_agnostic : bool, "all" or any subset of ["dataset_description.json", "CHANGES", "README", "LICENSE"]
```

Comment:
You should consider making include_modality_agnostic=["dataset_description.json"] the default, 
considering pyAFQ's API requires it.

Reply:
Done.